ts/nni_manager/test/common/gpu_scheduler.ts

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import assert from 'assert/strict';
```

Comment:
Suggest adding a test on real GPU.

Maybe on full test HPO.

Reply:
Er, maybe later as a part of integration test.
I'm scared of those environments :(

Reply:
I don't agree it's integration test. Technically we don't have integration tests now. We only have full tests for HPO and training services. I don't think it's part of the training service. I think at least the GPU scheduler deserves a dedicated test rather than only be tested in an end-to-end experiment.

It's acceptable that it will be done in a separate PR.

Reply:
The GPU scheduler will become a separate daemon process. It will have its own entry point file alongside `main.ts`. So I will add GPU integration tests for the daemon, and use mocked unit tests for the library (the class).